### PR TITLE
Improve some Haddock in BlockFetch

### PR DESCRIPTION
**Pull request by nfrisby**
Updated some stale Haddock references. Improved wording and added details to `FetchDecline` Haddock.

### CODE REVIEW

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs [hunk 1]
```diff
@@ -89,77 +89,144 @@ type FetchDecision result = Either FetchDecline result
...
+--   chain from other peers (or from the node's own block forge). Because the
+--   node's current selection is influenced by what blocks other peers have
+--   recently served (or it recently minted), this reason reflects that peers
+--   /indirectly/ compete by serving as long of a chain as possible and as
+--   promptly as possible. When the tips of the peers' selections are all
+--   within their respective forecast horizons (see
+--   'Ouroboros.Consensus.Ledger.SupportsProtocol.ledgerViewForecastAt'), then
+--   the length of their candidate chains will typically be the length of their
+--   selections, since ChainSync should severely outpace BlockFetch (both much
+--   smaller payloads and also much cheaper validation). But if one or more of
```
Comment (amesgen) [inline ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:120]: By "validation", do you mean something like parsing the block CBOR in the context of BlockFetch?

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs [hunk 2]
```diff
@@ -89,77 +89,144 @@ type FetchDecision result = Either FetchDecline result
...
-     -- | The chain does not intersects with our current chain.
+     -- | Switching to this peer's candidate chain would require rolling back
+     -- more than @K@ blocks.
+     --
+     -- Clarification: this constructor name is technically nonsensical. Every
+     -- pair of valid chains intersects, if only at the genesis block! Even so,
+     -- the current name is sufficiently clear for those with some intuition
+     -- for the latent assumptions of ChainSync, BlockFetch, Praos, etc. But a
+     -- more precise name such as @FetchDeclineIntersectionTooDeep@ would rely
+     -- less on familiarity with that context.
```
[the remark this answers is not in the document]
Comment (nfrisby) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:163]: I considered it, but it'd require updating downstream code too, etc. So I thought I'd leave that for you to decide later :)

But if you're already fine with that, I could make that change in this PR.

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs (hunk 1 above)
Comment (nfrisby) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:120]: Thanks, I reworded this.

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs [hunk 3]
```diff
@@ -89,77 +89,144 @@ type FetchDecision result = Either FetchDecline result
...
    | FetchDeclineChainNoIntersection
 
-     -- | The chain has already been fetched.
+     -- | Every block on this peer's candidate chain has already been fetched.
      --
    | FetchDeclineAlreadyFetched
 
-     -- | The chain is already being downloaded from that peer.
+     -- | This peer's candidate chain is has already been requested from this
+     -- peer.
```
[the remark this answers is not in the document]
Comment (nfrisby) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:172]: Thanks. Done.

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs [hunk 4]
```diff
@@ -89,77 +89,144 @@ type FetchDecision result = Either FetchDecline result
 
 -- | All the various reasons we can decide not to fetch blocks from a peer.
 --
+-- It is worth highlighting which of these reasons result from competition
+-- among upstream peers.
+--
+-- * 'FetchDeclineInFlightOtherPeer': decline this peer because all the
+--   unfetched blocks of its candidate chain have already been requested
+--   fetched from other peers. This reason reflects the least-consequential
```
[the remark this answers is not in the document]
Comment (nfrisby) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:97]: Thanks. Done.

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (nfrisby) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:163]: Just to be clear: I'd rather not block this PR, if there's any delay in this decision.

[the remark this answers is not in the document]
Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:165]: That's ok, you can do that in a separate PR, or open an issue for us.

[the remark this answers is not in the document]
Comment (nfrisby) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:165]: Thanks. I created https://github.com/input-output-hk/ouroboros-network/issues/4541

**Comment by nfrisby** [comment]
Status update: I think this PR is ready for merge, once it gets an Approval.

**Review by coot — APPROVED** [review]
LGTM

**Comment by coot** [comment]
Thanks @amesgen & @nfrisby :) 

bors merge

**Pull request merged**